src/foam/nanos/approval/ApprovableAwareDAO.js

```diff
@@ -173,7 +173,8 @@ foam.CLASS({
         if ( lifecycleObj.getLifecycleState() == LifecycleState.PENDING && currentObjectInDAO == null ){
           lifecycleObj.setLifecycleState(LifecycleState.ACTIVE);
         }
-        return super.put_(x,obj);
+        // only system/admin should have permission to change lifecycle state
+        return super.put_(getX(), obj);
```

Comment:
I don't think you want to change the context right in the middle of a put operation to the system context .. won't this affect all fo the operations that require the "user" or "agent" from the context, like CreatedBy/ModifiedBy?